src/test/java/org/scijava/search/module/HelpSearchActionTest.java

```diff
@@ -0,0 +1,210 @@
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+package org.scijava.search.module;
+
+import static org.junit.Assert.*;
+
+import java.lang.reflect.Field;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.scijava.Context;
+import org.scijava.app.AppService;
+import org.scijava.command.CommandInfo;
+import org.scijava.module.ModuleInfo;
+import org.scijava.platform.AppEventService;
+import org.scijava.platform.Platform;
+import org.scijava.platform.PlatformService;
+import org.scijava.plugin.PluginInfo;
+import org.scijava.search.DefaultSearchAction;
+import org.scijava.search.SearchAction;
+import org.scijava.search.SearchActionFactory;
+import org.scijava.search.SearchResult;
+
+/**
+ * Tests {@link HelpSearchActionFactory}
+ *
+ * @author Gabriel Selzer
+ */
+public class HelpSearchActionTest {
+
+	private AppService appService;
+	private SearchActionFactory factory = new HelpSearchActionFactory();
+
+	@Before
+	public void init() {
+		Context context = new Context();
+		appService = context.getService(AppService.class);
+		factory = new HelpSearchActionFactory();
+	}
+
+	@Test
+	public void testSupports() {
+		final SearchResult good = createTestResult();
+		assertTrue(factory.supports(good));
+		final SearchResult bad = new SearchResult() {
+
+			@Override
+			public String name() {
+				return null;
+			}
+
+			@Override
+			public String iconPath() {
+				return null;
+			}
+
+			@Override
+			public Map<String, String> properties() {
+				return null;
+			}
+		};
+		assertFalse(factory.supports(bad));
+	}
+
+	@Test
+	public void testCreate() {
+		final SearchResult result = createTestResult();
+		SearchAction action = factory.create(result);
+		assertTrue(action instanceof DefaultSearchAction);
+		assertEquals("Help", action.toString());
+	}
+
+	@Test
+	public void testHelp() throws NoSuchFieldException, IllegalAccessException {
+		final SearchResult result = createTestResult();
+		final MockPlatformService mockService = new MockPlatformService();
+		Field platformService = factory.getClass().getDeclaredField(
```

Comment:
A nicer way without reflection would be to add a package-private `setPlatformService(PlatformService)` method to `HelpSearchActionFactory`, and change the declared type of the `factory` field accordingly to be more specific.

Reply:
I am hesitant to do that seeing as how our only use case is for testing `HelpSearchActionFactory`...

Reply:
I am concerned the reflection approach will break in a future (or even current) version of Java. Whereas package-private methods have historically been the way to let unit tests manipulate classes internally in a privileged way. I agree it is suboptimal in that it changes the bytecode of the class to have this extra useless internal method.

Reply:
But I guess it's not a blocker to merging the PR. If it breaks later, we can fix it later.

Reply:
I guess I prefer the old "if it ain't broke, don't fix it".

You are right that there are concerns about the reflection approach, but I'd rather wait until it becomes a problem than add API (and add a second way to manipulate a private field).